spec/stud/interval_spec.rb

```diff
@@ -1,8 +1,19 @@
 require "stud/interval"
 require "spec_env" # from the top level spec/ directory
 
+module Stud
+  def reset!(target = Thread.current)
```

Comment:
I think this shouldn't be only just for tests, let's say we have a `threadpool` to reuse our threads and we stop a specific interval. When we later reuse the thread the `STUD_STOP_REQUESTED` will be already set and it will make all the newly created interval stop. So I think we should set that value to `NIL` once the interval is stopped.

We don't use threadpool everywhere but I do use them in the lumberjack input and probably will make the same patch for the tcp/udp inputs.